framework/contracts/native/ica-client/README.md

```diff
@@ -0,0 +1,39 @@
+# Abstract ICA Client
+
+The ICA client contract provides an interface for interacting with Interchain Accounts (ICAs) on different blockchains using the IBC protocol.
```

Comment:
What we have with the ibc-client and host is something similar to the "controlled note" because they form a protocol built upon Polytone as a messaging layer. 

Is the ICA-client going to be separate from the ibc-client and host and be more of a "protocol-level" contract? Asking because if the user calls through Polytone with this they'll have a separate Polytone Proxy on the other chain. Or is the ibc-client ultimately going to use this?

Reply:
The ibc-client could use this to interact with its polytone instance yes (if we added polytone as an ICA provider)

This is basically an ICA protocol adapter.

Reply:
Gotcha. So are we imagining that we'll create a new `evm-ibc-client` or execute EVM ica actions through this one? 

I suppose it relates to the question whether we'll support "ibc-host"-like functionality on the EVM side.

Reply:
We execute EVM ICA actions "through" this adapter. This contract only constructs messages based on a query. 

ibc-host-like functionality is a different game altogether and would broaden the scope of our EVM work significantly. Ideally we update the proxy implementation to use some available smart-account solution. 

It's too easy to assume that certain patterns in CW also make sense in Solidity. The complexity of building an abstract-like setup on EVM (which would be a requirement to supporting ibc-host functionality) is probably an order of magnitude harder than CW. 

I really think we should keep playing on our strengths (CW) and keep the EVM side either as minimal as possible (with some additional runtime overhead) or finding an EVM SA provider that is willing to play this game with us.